src/NHibernate/Engine/CollectionEntry.cs

```diff
@@ -361,6 +363,26 @@ public void AfterAction(IPersistentCollection collection)
 			collection.PostAction();
 		}
 
+		public void AfterAction(IPersistentCollection collection, ISessionImplementor session)
```

Comment:
Instead of creating this method you could just set CurrentKey to Action's Key before calling this method.

Reply:
Yes it would work too, but that means fixing this at three places instead of fixing it at one place. And if some change introduces later a new action class, it may forget to handle this. Adding this method is more robust.

By the way, the suggestions are currently not visible anywhere else than my emails, so I could not accept them. That is maybe because you are the author of the PR, and Github has maybe not correctly handled the case of suggestions made by the PR's author on a commit pushed by someone else.

Reply:
> the suggestions are currently not visible anywhere

I deleted them.

I was thinking maybe the key resolution might be done in the wrong place.